rs/execution_environment/src/execution_environment.rs

```diff
@@ -3221,20 +3214,17 @@ impl ExecutionEnvironment {
     /// Aborts paused execution in the given state.
     pub fn abort_canister(&self, canister: &mut CanisterState, log: &ReplicaLogger) {
         if !canister.system_state.task_queue.is_empty() {
-            let paused_tasks = canister.system_state.task_queue.get_all_paused_tasks();
+            if let Some(paused_task) = canister.system_state.task_queue.get_paused_task() {
+                self.metrics.executions_aborted.inc();
+                // TODO: EXC-1730 if `PausedExecutionRegistry` becomes local we can abort
+                // paused execution on the canister without requesting ID from TaskQueue.
+                let aborted_task = self.abort_paused_execution_and_return_task(paused_task, log);
 
-            self.metrics
-                .executions_aborted
-                .inc_by(paused_tasks.len().try_into().unwrap());
-
-            // TODO: EXC-1730 if `PausedExecutionRegistry` becomes local
-            // we can abort all executions on it without `paused_tasks`.
-            let aborted_tasks = self.abort_paused_executions_and_return_tasks(paused_tasks, log);
-
-            canister
-                .system_state
-                .task_queue
-                .replace_paused_with_aborted_tasks(aborted_tasks);
+                canister
+                    .system_state
+                    .task_queue
+                    .replace_paused_with_aborted_task(aborted_task);
```

Comment:
Nit: would it be cleaner to merge the functionality of `get_paused_task`, `abort_paused_execution_and_return_task` and `replace_paused_with_aborted_task` in one function?

Reply:
It would be great, but I am not aware of the way to do it, since we first need PausedExecutionId from TaskQueue, then we need to get that execution from PausedExecutionRegistry which is global, and using that info we can create AbortedTask, which eventually finishes on TaskQueue. That was the reason I noticed that PausedExecutionRegistry was global as we discussed.